src/cljs/athens/views/blocks.cljs

```diff
@@ -221,7 +223,18 @@
 (defn on-change
   [value uid]
   ;; (prn "ONCHANGE" value)
-  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]]))
+  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]])
+  ;; automatically add non-existent pages
+  ;; TODO: delete pages that are no longer connected to anything else
+  (parse/transform {:page-link (fn [& title]
+                                 (if (nil? (db/search-exact-node-title (apply + title)))
+                                   (let [uid (gen-block-uid)]
+                                     (dispatch [:transact [{:node/title     (apply + title)
```

Comment:
If you're going to use `(apply + title)` up to three times, I'd just define it in a `let` binding at the top-level.

Reply:
Fixed